bigbluebutton-html5/imports/ui/components/chat/chat-dropdown/component.jsx

```diff
@@ -0,0 +1,159 @@
+import DropdownContent from '/imports/ui/components/dropdown/content/component';
+import DropdownList from '/imports/ui/components/dropdown/list/component';
+import DropdownListItem from '/imports/ui/components/dropdown/list/item/component';
+import { makeCall } from '/imports/ui/services/api';
+import Auth from '/imports/ui/services/auth';
+import Chats from '/imports/api/2.0/chat';
+import Acl from '/imports/startup/acl';
+import ChatService from './../service';
+import styles from './styles';
+
+const intlMessages = defineMessages({
+  clear: {
+    id: 'app.chat.dropdown.clear',
+    description: 'Clear button label',
+  },
+  save: {
+    id: 'app.chat.dropdown.save',
+    description: 'Clear button label',
+  },
+  copy: {
+    id: 'app.chat.dropdown.copy',
+    description: 'Copy button label',
+  },
+  options: {
+    id: 'app.chat.dropdown.options',
+    description: 'Chat Options',
+  },
+});
+
+const CHAT_CONFIG = Meteor.settings.public.chat;
+const PUBLIC_CHAT_USERNAME = CHAT_CONFIG.public_username;
+const SYSTEM_CHAT_TYPE = CHAT_CONFIG.type_system;
+
+class ChatDropdown extends Component {
+
+  static getPublicMessages() {
+    const publicMessages = Chats.find({
+      'message.toUsername': { $in: [PUBLIC_CHAT_USERNAME, SYSTEM_CHAT_TYPE] },
+    }, {
+      sort: ['message.fromTime'],
+    }).fetch();
+
+    return ChatService.exportChat(publicMessages).join('\n');
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isSettingOpen: false,
+    };
+
+    this.onActionsShow = this.onActionsShow.bind(this);
+    this.onActionsHide = this.onActionsHide.bind(this);
+  }
+
+  componentDidMount() {
+    this.clipboard = new Clipboard('#clipboardButton', {
+      text: () => ChatDropdown.getPublicMessages(),
+    });
+  }
+
+  componentWillUnmount() {
+    this.clipboard.destroy();
+  }
+
+  onActionsShow() {
+    this.setState({
+      isSettingOpen: true,
+    });
+  }
+
+  onActionsHide() {
+    this.setState({
+      isSettingOpen: false,
+    });
+  }
+
+  getAvailableActions() {
+    const { intl } = this.props;
+
+    const clearIcon = 'delete';
+    const saveIcon = 'save_notes';
+    const copyIcon = 'copy';
+
+    return _.compact([
+      (<DropdownListItem
+        icon={saveIcon}
+        label={intl.formatMessage(intlMessages.save)}
+        key={_.uniqueId('action-item-')}
+        onClick={() => {
+          const link = document.createElement('a');
+          const mimeType = 'text/plain';
+
+          link.setAttribute('download', 'chat.txt');
+          link.setAttribute('href', `data: ${mimeType} ;charset=utf-8, ${encodeURIComponent(ChatDropdown.getPublicMessages())}`);
+          link.click();
+        }}
+      />),
+      (<DropdownListItem
+        icon={copyIcon}
+        id="clipboardButton"
+        label={intl.formatMessage(intlMessages.copy)}
+        key={_.uniqueId('action-item-')}
+      />),
+      (Acl.can('methods.clearPublicChatHistory', Auth.credentials) ?
```

Comment:
No, it shouldn't, since we have dynamic roles, if we check for `role === moderator`, the ACL won't work.

The ideia is, if you for example, add inside the `presenter` role the `clearPublicChatHistory`, it should be allow.